test/integration/federation/framework/api.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/pborman/uuid"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	restclient "k8s.io/client-go/rest"
+	federationclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_clientset"
+	"k8s.io/kubernetes/federation/cmd/federation-apiserver/app"
+	"k8s.io/kubernetes/federation/cmd/federation-apiserver/app/options"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+const PortNotSet = 0
+
+func getRunOptions() *options.ServerRunOptions {
+	r := options.NewServerRunOptions()
+	r.Etcd.StorageConfig.ServerList = []string{framework.GetEtcdURLFromEnv()}
+	// Use a unique prefix to ensure isolation from other tests using the same etcd instance
+	r.Etcd.StorageConfig.Prefix = uuid.New()
+	// Disable secure serving
+	r.SecureServing.ServingOptions.BindPort = 0
+	// The insecure port will be set to an ephemeral port before starting the api
+	r.InsecureServing.BindPort = PortNotSet
+	return r
+}
+
+// FederationAPIFixture manages a federation api server
+type FederationAPIFixture struct {
```

Comment:
This is a test fixture:

https://en.wikipedia.org/wiki/Test_fixture

I think it makes sense to identify it as such rather than obscuring it with new and different names.